libs/partners/cohere/langchain_cohere/chat_models.py

```diff
@@ -60,19 +58,17 @@ def get_cohere_chat_request(
     Returns:
         The request for the Cohere chat API.
     """
-    documents = (
-        None
-        if "source_documents" not in kwargs
-        else [
-            {
-                "snippet": doc.page_content,
-                "id": doc.metadata.get("id") or f"doc-{str(i)}",
-            }
-            for i, doc in enumerate(kwargs["source_documents"])
-        ]
-    )
-    kwargs.pop("source_documents", None)
-    maybe_connectors = connectors if documents is None else None
+    additional_kwargs = messages[-1].additional_kwargs
+
+    # cohere SDK will fail loudly if both connectors and documents are provided
+    documents = [
```

Comment:
What are your thoughts on adding `documents` and `connectors` to `ChatCohere` ?

Reply:
Hmm I don't think that would be needed. The sense I have been getting from the LC people is that they like having such inference based parameters (documents, connectors) on a "per message" basis vs generic ChatModel configurations. That way with new instantiations of eg. `HumanMessage` you can configure which parameters you want to include. So that means that in a conversation history you can have multiple turns where say first turn used documents, second turn used connectors and third turn didn't use anything.

Reply:
That makes sense!